### ISSUE TITLE
Pprotect doesn't call is function after reset

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#4912]
This issue is identical to #4658 and it will be fixed by #4659.

**Let's keep this one open too, so it can be more easily be traced back.**

## Environment

* SuperCollider version:  at current commit adaddb73bd40635d29678a121fb4e2ecbd2a4534

## Steps to reproduce

The short version shows that the function isn't called after reset:
```supercollider
a = Pfunc { Error("dummy").throw };
p = PauseStream(Pprotect(a, { "----> Pprotect function called".postln; p.streamError }).asStream);
p.play; // ok, throw error
p.reset;
p.play; // ok, throw again, but  Pprotect function was _not_ called
```

The longer version shows how it affects protection of `PauseStream`
```supercollider
a = Pfunc { Error("dummy").throw };
p = PauseStream(Pprotect(a, { "----> Pprotect function called".postln; p.streamError }).asStream);
p.play; // ok, throw error
p.reset;
p.play; // ok, throw again, but  Pprotect function was _not_ called
p.reset;
p.play; // suddenly: "already playing", no throw.
p.streamError;  // set it by hand
p.play; // ok, throw again
```

And this shows how it leads to #4658.

```supercollider
(
Tdef(\pp, {
		Error("dummy").throw
});
)

Tdef(\pp).play;
Tdef(\pp).play; // [ streamHasEnded, true, wasStopped, false, isPlaying, false ]
Tdef(\pp).play; // third time it doesn't work anymore. [ streamHasEnded, false, wasStopped, false, isPlaying, true ]
Tdef(\pp).stop; // reset it
Tdef(\pp).play; // again it works
```

## Expected vs. actual behavior

We expect the function to be called after reset and the `Tdef` to just play after an error. This is an annoying, "user-facing" error.

**Comment by telephon** [comment]
fixed by  #4659

**Comment by brianlheim** [comment]
thanks @telephon !
